atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/SqliteOracleAdapter.java

```diff
@@ -0,0 +1,309 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.dbkvs.impl.oracle;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.endsWith;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.spy;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.keyvalue.api.Namespace;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.dbkvs.ImmutableOracleDdlConfig;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleDdlConfig;
+import com.palantir.atlasdb.keyvalue.dbkvs.OracleTableNameGetter;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.OverflowMigrationState;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.SimpleTimedSqlConnectionSupplier;
+import com.palantir.atlasdb.keyvalue.dbkvs.impl.SqlConnectionSupplier;
+import com.palantir.common.base.FunctionCheckedException;
+import com.palantir.exception.PalantirSqlException;
+import com.palantir.nexus.db.ConnectionSupplier;
+import com.palantir.nexus.db.sql.SqlConnection;
+import com.palantir.util.file.TempFileUtils;
+import java.io.File;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Collection;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.immutables.value.Value;
+
+final class SqliteOracleAdapter implements ConnectionSupplier {
+    public static final TableReference METADATA_TABLE = AtlasDbConstants.DEFAULT_ORACLE_METADATA_TABLE;
+    private final File sqliteFileLocation;
+
+    SqliteOracleAdapter() throws IOException {
+        sqliteFileLocation = TempFileUtils.createTempFile("sqlite", ".db");
+    }
+
+    @Override
+    public Connection get() throws PalantirSqlException {
+        try {
+            // As with all connection supplier things, clients are expected to close connections.
+            return applyPragma(DriverManager.getConnection(getConnectionUrl()));
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```

Comment:
This seems... suspicious given the parent class says that a `PalantirSqlException` is thrown on a failure to get the connection. Should this not be `PalantirSqlException.create(e)`?

Reply:
Yeah, that was silly. I copied my implementation from runWithConnection (where I don't really care about dealing with checked exceptions). Have fixed